block-padding/src/lib.rs

```diff
@@ -64,7 +65,7 @@ pub trait Padding {
     /// Unpad data in the `blocks`.
     ///
     /// Returns `Err(UnpadError)` if the block contains malformed padding.
-    fn unpad_blocks<BlockSize: ArraySize>(
+    fn unpad_blocks<const N: usize, BlockSize: ArraySize<ArrayType<u8> = [u8; N]>>(
```

Comment:
I should've seen it coming...

This bound is quite annoying ad will leak in the downstream crates, but I guess it should not be a problem in practice.

Reply:
Yeah, these bounds are definitely annoying. They can be avoided at the cost of more unsafe code, which is something which has been done with other parts of `hybrid-array`

Reply:
@tarcieri 
WDYT? I see 3 options:
1) Keep the old simple bound and `unsafe`
2) Tolerate the new bound
3) Try to relax the bounds in `hybrid-array` using `unsafe`